metal/csr.h

```diff
@@ -18,15 +18,22 @@
  * @param crs Register label or hex value offset to read from
  * @param value Variable name of uintprt_t type to get the value
  */
+
+#ifdef __ASSEMBLY__
+#define __ASM_STR(x) x
+#else
+#define __ASM_STR(x) #x
+#endif
+
```

Comment:
Why do we need to have this __ASM_STR(x) define for ASSEMBLY when no commit in assembly in this PR use it ?

Reply:
I'm not sure if these APIs are only for freedom-metal internal use or they are used for user as well. I saw that some example programs in `software/` directory invoke `METAL_CPU_GET_CSR/METAL_CPU_SET_CSR`, if they are also used for user, we might not know how people use it, but if they are only used for freedom-metal internal, it can be dropped as you mentioned. Could you please help me to point out what is the expectation on those and whether we need to remove the __ASSEMBLY__ part. Thanks.